pkg/tools/nanoid/generator.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2024 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nanoid is a tiny, unique string ID generator. Original JavaScript implementation: https://github.com/ai/nanoid
+package nanoid
+
+import (
+	"crypto/rand"
+	"math"
+	"math/bits"
+)
+
+const (
+	// DefaultAlphabet is the default alphabet for the generator which can be used to generate kernel interface names
+	DefaultAlphabet = "!\"#$&'()*+,-.012456789;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
+)
+
+type generatorOpts struct {
+	alphabet string
+}
+
+// Option represents options for the string generator
+type Option func(o *generatorOpts)
+
+// WithAlphabet sets a custom alphabet for the  generator
+func WithAlphabet(alphabet string) Option {
+	return func(o *generatorOpts) {
+		o.alphabet = alphabet
+	}
+}
+
+func generateRandomBuffer(step int) ([]byte, error) {
+	buffer := make([]byte, step)
+	if _, err := rand.Read(buffer); err != nil {
```

Comment:
Could we use https://pkg.go.dev/crypto/rand?

Original Nanoid based on the crypto module